src/js/stores/DCOSStore.js

```diff
@@ -120,7 +123,13 @@ class DCOSStore extends EventEmitter {
     this.data.marathon.deploymentsList = deploymentsList
       .mapItems(function (deployment) {
         let ids = deployment.getAffectedServiceIds();
-        let services = ids.map(serviceTree.findItemById.bind(serviceTree));
+        let services = ids.reduce(function (memo, id) {
+          let service = serviceTree.findItemById(id);
```

Comment:
How about moving the data merging to the getter - similar the services? With that change you wouldn't need the `dataProcessed` conditions as you could return the deployment list whenever the needed data is present. You're maybe even able to provide a subset until all needed data is present. What do you think?